src/main/scala/chisel3/util/LoadMemoryTransform.scala

```diff
@@ -0,0 +1,249 @@
+// See LICENSE for license details.
+
+package chisel3.util
+
+import chisel3.MemBase
+import chisel3.core.{RunFirrtlTransform, annotate, ChiselAnnotation}
+import chisel3.internal.{Builder, InstanceId}
+import firrtl.annotations._
+import firrtl.ir.{Module => _, _}
+import firrtl.passes.Pass
+import firrtl.passes.memlib.DefAnnotatedMemory
+import firrtl.transforms.BlackBoxInlineAnno
+import firrtl.{CircuitForm, CircuitState, LowForm, Transform, VerilogEmitter, WDefInstance}
+
+import scala.collection.mutable
+import scala.util.matching.Regex
+
+/**
+  * chisel implementation for load memory
+  * @param target        memory to load
+  * @param fileName      name of input file
+  * @param hexOrBinary   use $readmemh or $readmemb
+  */
+case class ChiselLoadMemoryAnnotation(
+  target     : InstanceId,
+  fileName   : String,
```

Comment:
Rather than filename, I would call this `prefix` and also add a field for file extension since people may have their own preferred file extension for these kinds of files.

Reply:
This is now the filename, if aggregate memories are involved I use the filename as a template and
insert the sub-element name before the suffix. So I think that makes filename the right name for the var.
There is wiki documentation that explains this to the user